### PR TITLE
Wire cache handlers in edge paths and add e2e regression coverage

### PR DESCRIPTION
**Pull request by ijjk**
### What?

This PR wires custom `cacheHandler` / `cacheHandlers` through the remaining edge entrypoints so edge app pages, edge app routes, edge pages SSR, middleware, and edge API routes can all see the configured handlers.

It also adds an e2e regression suite covering:
- pages-router ISR revalidation with a custom incremental cache handler
- app-router cache handlers with cache components enabled
- edge app page + edge app route wiring when cache components are disabled

### Why?

We already support custom cache handlers in non-edge paths, but several edge code paths were not forwarding that configuration into the generated entrypoints/runtime wrappers. That meant custom cache handlers could be skipped in edge rendering and edge route execution, and we did not have regression coverage around those cases.

### How?

- pass `cacheHandler` / `cacheHandlers` through the JS build entry plumbing for edge server entries
- inject cache handler imports/registration into the webpack edge templates and loaders
- mirror the same wiring in the Turbopack/Rust entry generation for edge app pages, app routes, pages SSR, and middleware
- update the edge route wrapper to initialize and register cache handlers before invoking the route module
- extend `next-taskless` template expansion with raw injection support so the generated edge templates can add imports and registration code
- add `test/e2e/cache-handlers-upstream-wiring` fixtures to cover pages/app, edge/non-edge, and revalidation behavior

### CODE REVIEW

File: packages/next/src/build/templates/edge-app-route.ts [hunk 1]
```diff
@@ -24,10 +27,15 @@ if (rscManifest && rscServerManifest) {
...
 export const ComponentMod = module
 
+const edgeCacheHandlers: any = {}
+// INJECT_RAW:edgeCacheHandlersRegistration
+
 const internalHandler: EdgeHandler = EdgeRouteModuleWrapper.wrap(
   module.routeModule,
   {
     page: 'VAR_PAGE',
+    cacheHandlers: edgeCacheHandlers,
```
Comment (ztanner) [inline packages/next/src/build/templates/edge-app-route.ts:37]: Cache Components doesn't support edge runtime and `cacheHandlers` only backs `use cache` so we can probably omit this right?

Comment (ijjk) [reply packages/next/src/build/templates/edge-app-route.ts:37]: This was wiring to make sure `revalidateTag` and `updateTag` still work in edge since that is valid to do without using cache components in edge also helps to wire everywhere for consistency

Comment (ztanner) [reply packages/next/src/build/templates/edge-app-route.ts:37]: What if there's an app that has a node-only `cacheHandlers` impl? Those edge bundles might fail now because the edge graph has to handle the node-only module right?

**Review by ztanner — APPROVED** [review]
Approved assuming the edge bundling concerns are acceptable
